src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -1751,6 +1763,8 @@ string dumpDagTest(BoundDagTest d)
         /// </summary>
         private sealed class DagState
         {
+            private static readonly ObjectPool<DagState> s_dagStatePool = new ObjectPool<DagState>(() => new DagState());
```

Comment:
Note: it might make sense to figure out a reasonable larger default here so that large switches don't blow the entire pool out and only return a small subset of dag states back to it.